### PR TITLE
Feat: Folders - Add Bottom Navigation

### PR DESCRIPTION
**Pull request by gizemb**
### Issues

Add Bottom Navigation that enables switching between Recent Conversations view & soon-to-be-added Foldered Conversations view.

### Testing

Page navigations, especially special cases like Pending Conversation page, etc. should be tested.

#### APK
[Download build #114](http://10.10.124.11:8080/job/Pull%20Request%20Builder/114/artifact/build/artifact/wire-dev-PR2332-114.apk)
[Download build #117](http://10.10.124.11:8080/job/Pull%20Request%20Builder/117/artifact/build/artifact/wire-dev-PR2332-117.apk)
[Download build #118](http://10.10.124.11:8080/job/Pull%20Request%20Builder/118/artifact/build/artifact/wire-dev-PR2332-118.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala [hunk 1]
```diff
@@ -80,10 +85,23 @@ class ConversationListManagerFragment extends Fragment
...
+    convs <- z.convsStorage.contents
+  } yield {
+      convs.values.exists(c => c.archived && !c.hidden && !Set(Self, Unknown).contains(c.convType))
+  }
 
   private def stripToConversationList() = {
     pickUserController.hideUserProfile() // Hide possibly open self profile
-    if (pickUserController.hidePickUser()) navController.setLeftPage(Page.CONVERSATION_LIST, Tag) // Hide possibly open start ui
+    if (pickUserController.hidePickUser()) { // Hide possibly open start ui
+      returning(navController) { c => c.setLeftPage(c.getLastSelectedConversationPage, Tag) }
```
Comment (johnxnguyen) [inline app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala]: I see this line is duplicated many times, with the only difference being the tag that is passed in. Its a nice opportunity to express this with a method.

**Comment by johnxnguyen** [comment]
My changes also touched `ConversationListFragment`, but those changes are mostly clean ups. I think it would be easier to fix conflicts on my side, so I would merge this before merging my branch.

File: app/src/main/res/drawable/ic_folderview.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    android:width="357dp"
+    android:height="356.844dp"
+    android:viewportWidth="357"
+    android:viewportHeight="356.844">
+  <path
+      android:pathData="M307,262C307,262 234.845,262 167,262C167,262 79.675,354.279 78,356C75.362,358.711 63.017,354.491 63,349L63,262C62.974,262 62.946,262 62.92,262C54.735,262 50,262 50,262C22.386,262 -0,239.614 -0,212L-0,50C-0,22.386 22.386,-0 50,-0L307,-0C334.614,-0 357,22.386 357,50L357,212C357,239.614 334.614,262 307,262ZM235,203L258,203C266.284,203 273,196.284 273,188C273,179.716 266.284,173 258,173L235,173C226.716,173 220,179.716 220,188C220,196.284 226.716,203 235,203ZM198,117C198,101.536 185.464,89 170,89L124.5,89L119.166,74.157C115.617,64.282 104.717,57 94.5,57L69.5,57C59.283,57 51,64.282 51,74.157L51,89L51,175C51,190.464 63.536,203 79,203L170,203C185.464,203 198,190.464 198,175L198,117ZM283,57L235,57C226.716,57 220,63.716 220,72C220,80.284 226.716,87 235,87L283,87C291.284,87 298,80.284 298,72C298,63.716 291.284,57 283,57ZM310,115L235,115C226.716,115 220,121.716 220,130C220,138.284 226.716,145 235,145L310,145C318.284,145 325,138.284 325,130C325,121.716 318.284,115 310,115Z"
```
[the remark this answers is not in the document]
Comment (gizemb) [reply app/src/main/res/drawable/ic_folderview.xml]: this is an svg image. It produces this xml when imported.

File: app/src/main/res/menu/main_bottom_navigation_items.xml [hunk 3]
```diff
@@ -18,8 +18,17 @@
...
-             android:layout_width="match_parent"
-             android:layout_height="match_parent"
-    />
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:id="@+id/navigation_search"
+        android:title="@null"
+        android:icon="@drawable/search" />
+    <item android:id="@+id/navigation_recents"
+        android:title="@null"
+        android:icon="@drawable/ic_recentview" />
```
[the remark this answers is not in the document]
Comment (gizemb) [reply app/src/main/res/menu/main_bottom_navigation_items.xml]: how about just "recentview"?

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala (hunk 1 above)
Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala]: done

File: app/src/main/res/menu/main_bottom_navigation_items.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnxnguyen) [reply app/src/main/res/menu/main_bottom_navigation_items.xml]: Same for the folder id.

[the remark this answers is not in the document]
Comment (gizemb) [reply app/src/main/res/menu/main_bottom_navigation_items.xml]: done

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala (hunk 1 above)
Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala]: I decided to simplify this logic. No need to export it to the NavigationController.

**Pull request merged**